In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Avoid SettingWithCopyWarning
pd.options.mode.copy_on_write = True

# Data Preparation

This dataset is from the U.S. Department of Education College Scorecard: https://collegescorecard.ed.gov/data/. The College Scorecard project is designed to increase transparency, putting the power in the hands of
students and families to compare how well individual postsecondary institutions are preparing their
students to be successful. This data was provided to help students and families compare college costs
and outcomes as they weigh the tradeoffs of different colleges, accounting for their own needs and
educational goals. 

The data contains real records of institutional data with over 6400 observations.

In [11]:
# Loading up the dataset and setting the datatype of "OPEID" and "OPEID6" column to string
institution_data = pd.read_csv('../data/Most-Recent-Cohorts-Institution.csv', dtype={"OPEID": str, "OPEID6": str}, low_memory=False)

In [12]:
# Make a copy of the original dataset
institution_data_orig = institution_data.copy()

## 1. Initial inspection

In [13]:
# Inspect the dataframe
institution_data

UNITID     OPEID  OPEID6  \
0       100654  00100200  001002   
1       100663  00105200  001052   
2       100690  02503400  025034   
3       100706  00105500  001055   
4       100724  00100500  001005   
...        ...       ...     ...   
6479  49178301  04270802  042708   
6480  49425001  02609404  026094   
6481  49501301  04247201  042472   
6482  49501302  04247202  042472   
6483  49664501  04285601  042856   

                                                 INSTNM         CITY STABBR  \
0                              Alabama A & M University       Normal     AL   
1                   University of Alabama at Birmingham   Birmingham     AL   
2                                    Amridge University   Montgomery     AL   
3                   University of Alabama in Huntsville   Huntsville     AL   
4                              Alabama State University   Montgomery     AL   
...                                                 ...          ...    ...   
6479                   Wilton Simpson Technical College  Brooksville     FL   
6480      Valley College - Fairlawn - School of Nursing     Fairlawn     OH   
6481  Western Maricopa Education Center - Southwest ...      Buckeye     AZ   
6482  Western Maricopa Education Center - Northeast ...      Phoenix     AZ   
6483  Burlington County Institute of Technology - Ad...      Medford     NJ   

             ZIP                                       ACCREDAGENCY  \
0          35762  Southern Association of Colleges and Schools C...   
1     35294-0110  Southern Association of Colleges and Schools C...   
2     36117-3553  Southern Association of Colleges and Schools C...   
3          35899  Southern Association of Colleges and Schools C...   
4     36104-0271  Southern Association of Colleges and Schools C...   
...          ...                                                ...   
6479   346134904                  Council on Occupational Education   
6480   443333631  Accrediting Commission of Career Schools and C...   
6481  85326-5705                  Council on Occupational Education   
6482  85027-0000                  Council on Occupational Education   
6483   080550000    Middle States Commission on Secondary Education   

                                      INSTURL  \
0                               www.aamu.edu/   
1                        https://www.uab.edu/   
2          https://www.amridgeuniversity.edu/   
3                                www.uah.edu/   
4                              www.alasu.edu/   
...                                       ...   
6479  https://hernandoschools.org/schools/stc   
6480                  https://www.valley.edu/   
6481       https://west-mec.edu/findyourhappy   
6482       https://west-mec.edu/findyourhappy   
6483          https://www.bcit.cc/bcitadulted   

                                                 NPCURL  ...  \
0     www.aamu.edu/admissions-aid/tuition-fees/net-p...  ...   
1     https://tcc.ruffalonl.com/University of Alabam...  ...   
2              https://www2.amridgeuniversity.edu:9091/  ...   
3                                       finaid.uah.edu/  ...   
4     www.alasu.edu/cost-aid/tuition-costs/net-price...  ...   
...                                                 ...  ...   
6479                                                NaN  ...   
6480                                                NaN  ...   
6481                                                NaN  ...   
6482                                                NaN  ...   
6483                                                NaN  ...   

      COUNT_WNE_MALE0_P11  COUNT_WNE_MALE1_P11  GT_THRESHOLD_P11  \
0                   800.0                777.0            0.6250   
1                  1811.0               1157.0            0.7588   
2                    75.0                 67.0            0.5986   
3                   810.0                802.0            0.7810   
4                  1224.0               1049.0            0.5378   
...    

In [14]:
# Check the dimensions of the DataFrame
institution_data.shape

(6484, 3305)

In [15]:
# Info about dimensions and data types of columns
institution_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6484 entries, 0 to 6483
Columns: 3305 entries, UNITID to MD_EARN_WNE_MALE1_P11
dtypes: float64(916), int64(15), object(2374)
memory usage: 163.5+ MB


In [16]:
# Summary statistics
institution_data.describe()

UNITID      SCH_DEG         HCM2         MAIN    NUMBRANCH  \
count  6.484000e+03  6047.000000  6484.000000  6484.000000  6484.000000   
mean   2.261726e+06     1.960311     0.006169     0.788865     3.191703   
std    7.805306e+06     0.901947     0.078307     0.408146     6.809394   
min    1.006540e+05     1.000000     0.000000     0.000000     1.000000   
25%    1.742222e+05     1.000000     0.000000     1.000000     1.000000   
50%    2.288965e+05     2.000000     0.000000     1.000000     1.000000   
75%    4.576172e+05     3.000000     0.000000     1.000000     2.000000   
max    4.966450e+07     3.000000     1.000000     1.000000    52.000000   

           PREDDEG      HIGHDEG      CONTROL      ST_FIPS       REGION  ...  \
count  6484.000000  6484.000000  6484.000000  6484.000000  6484.000000  ...   
mean      1.836212     2.247224     2.057989    28.956354     4.642813  ...   
std       1.084154     1.381255     0.832232    16.925459     2.180387  ...   
min       0.000000     0.000000     1.000000     1.000000     0.000000  ...   
25%       1.000000     1.000000     1.000000    13.000000     3.000000  ...   
50%       2.000000     2.000000     2.000000    29.000000     5.000000  ...   
75%       3.000000     4.000000     3.000000    42.000000     6.000000  ...   
max       4.000000     4.000000     3.000000    78.000000     9.000000  ...   

       COUNT_WNE_MALE0_P11  COUNT_WNE_MALE1_P11  GT_THRESHOLD_P11  \
count          4894.000000          4285.000000       4712.000000   
mean           1455.479158          1107.202100          0.597479   
std            5179.034533          3232.508489          0.170466   
min              16.000000            16.000000          0.128300   
25%             138.000000           120.000000          0.475175   
50%             443.000000           350.000000          0.607500   
75%            1315.750000          1049.000000          0.733500   
max          143629.000000         72022.000000          0.955900   

       MD_EARN_WNE_INC1_P11  MD_EARN_WNE_INC2_P11  MD_EARN_WNE_INC3_P11  \
count           4931.000000           4468.000000           3550.000000   
mean           38198.777124          46049.203894          52479.307324   
std            14797.971002          14409.067180          15652.851538   
min            11508.000000          12410.000000          14098.000000   
25%            28025.500000          37485.000000          43087.000000   
50%            35237.000000          44229.000000          50796.000000   
75%            45039.500000          52847.000000          60273.500000   
max           130914.000000         134484.000000         139763.000000   

       MD_EARN_WNE_INDEP0_P11  MD_EARN_WNE_INDEP1_P11  MD_EARN_WNE_MALE0_P11  \
count             4675.000000             4690.000000            4894.000000   
mean             41993.114652            40284.135821           38231.901921   
std              15479.329063            15858.913214           15018.898024   
min               9834.000000             9978.000000           10586.000000   
25%              31613.000000            28948.250000           27726.250000   
50%              39319.000000            37610.500000           35188.000000   
75%              49587.000000            48274.750000           45173.250000   
max             128900.000000           155413.000000          126750.000000   

       MD_EARN_WNE_MALE1_P11  
count            4285.000000  
mean            49084.291015  
std             17319.894343  
min              8364.000000  
25%             38522.000000  
50%             46525.000000  
75%             56888.000000  
max            248999.000000  

[8 rows x 931 columns]

In [17]:
# Data types of columns
institution_data.dtypes

UNITID                      int64
OPEID                      object
OPEID6                     object
INSTNM                     object
CITY                       object
                           ...   
MD_EARN_WNE_INC3_P11      float64
MD_EARN_WNE_INDEP0_P11    float64
MD_EARN_WNE_INDEP1_P11    float64
MD_EARN_WNE_MALE0_P11     float64
MD_EARN_WNE_MALE1_P11     float64
Length: 3305, dtype: object

We have 3305 entries, most of which are object datatypes. We know that most of these entries are not important for our research, so we'll start removing unwanted entries.

## 2. Selecting / removing columns

In [ ]:
# Removing unnecessary variables

institution_data = institution_data.loc[:, ~dirty_data.columns.str.contains("POOL", case=False)]
institution_data = institution_data.loc[:, ~dirty_data.columns.str.contains("3YR", case=False)]
institution_data = institution_data.loc[:, ~dirty_data.columns.str.contains("4YR", case=False)]
institution_data = institution_data.loc[:, ~dirty_data.columns.str.contains("5YR", case=False)]
institution_data = institution_data.loc[:, ~dirty_data.columns.str.contains("YR2", case = False)]
institution_data = institution_data.loc[:, ~dirty_data.columns.str.contains("YR3", case=False)]
institution_data = institution_data.loc[:, ~dirty_data.columns.str.contains("YR4", case=False)]
institution_data = institution_data.loc[:, ~dirty_data.columns.str.contains("YR6", case=False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("YR8", case=False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("DBRR", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("PLUS", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("OMEN", case = False) | dirty_data.columns.str.contains("WOMENONLY", case=False) |
                            dirty_data.columns.str.contains("UGDS_WOMEN", case=False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("OMAW", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("GT", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("DCS", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("MID", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("25", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("75", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("P6", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("P7", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("P8", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("P9", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("P11", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("SUPP", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("PROG", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("OTHER", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("PCIP", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("CIP", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("ACCRED", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("LOCALE", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("MTHCMP", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("SD", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("CDR", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("CNTOVER", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("OTHEREXPENSE", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("LPSTAFFORD", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("12MN", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("URL", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("D150", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("D200", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("C200", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("D100", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("DTRANS", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("RPY", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("REPAY", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("SEPAR", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("APPL_SCH_PCT", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("FSEND", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("OMACHT", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("048", case = False)]
dirty_data = dirty_data.loc[:, ~dirty_data.columns.str.contains("CUML_DEBT", case = False)]



dirty_data = dirty_data.drop(columns=["ZIP", "SCH_DEG", "SCHTYPE", "HCM2", "CCBASIC", "UG", "PPTUG_EF2", "PFTFTUG1_EF", "ST_FIPS", "COUNT_ED", "AGE_ENTRY_SQ",
                                      "GRAD_DEBT_MDN10YR", "DEP_STAT_N", "PAR_ED_N", "APPL_SCH_N", "D_PCTPELL_PCTFLOAN", "T4APPROVALDATE", "UGNONDS", "OPEFLAG", "FEDSCHCD",
                                      "DOLPROVIDER", "MDCOMP_ALL", "MDCOST_ALL", "MDEARN_ALL", "UG_NRA", "UG_UNKN", "UG_WHITENH", "UG_BLACKNH", "UG_API",
                                      "UG_AIANOLD", "UG_HISPOLD", "UGDS_WHITENH", "UGDS_BLACKNH", "UGDS_API",
                                      "UGDS_AIANOLD", "UGDS_HISPOLD", "C150_4_WHITENH", "C150_4_BLACKNH", "C150_4_API", "C150_4_AIANOLD", "C150_4_HISPOLD", "C150_L4_WHITENH",
                                      "C150_L4_BLACKNH", "C150_L4_API", "C150_L4_AIANOLD", "C150_L4_HISPOLD", "MN_EARN_WNE_INDEP0_INC1_P10", "MN_EARN_WNE_INDEP1_P10",
                                      "MN_EARN_WNE_INDEP0_P10", "MN_EARN_WNE_INC1_P10", "MN_EARN_WNE_INC2_P10", "MN_EARN_WNE_INC3_P10", "MD_EARN_WNE_INC1_P10",
                                      "MD_EARN_WNE_INC2_P10", "MD_EARN_WNE_INC3_P10", "COUNT_WNE_INDEP0_P10", "COUNT_WNE_INDEP0_INC1_P10", "ADDR", "SATVR50", "SATMT50",
                                      "ACTCM50", "ACTEN50", "ACTMT50", "PRGMOFR", "PCT10_EARN_WNE_P10", "PCT90_EARN_WNE_P10", "COUNT_WNE_INDEP1_P10"])